pkg/hubble/getters.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hubble
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	hubbleProto "github.com/cilium/hubble/api/v1/flow"
+	hubbleV1 "github.com/cilium/hubble/pkg/api/v1"
+	hubbleIPCache "github.com/cilium/hubble/pkg/ipcache"
+	hubbleEndpoint "github.com/cilium/hubble/pkg/parser/endpoint"
+
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/endpointmanager"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/identity/cache"
+	"github.com/cilium/cilium/pkg/ipcache"
+	"github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/service"
+)
+
+// LocalIdentityGetter implements IdentityGetter interface from Hubble.
+type LocalIdentityGetter struct {
+	allocator *cache.CachingIdentityAllocator
+}
+
+// NewLocalIdentityGetter returns an initialized pointer to LocalIdentityGetter.
+func NewLocalIdentityGetter(allocator *cache.CachingIdentityAllocator) *LocalIdentityGetter {
+	return &LocalIdentityGetter{allocator: allocator}
+}
+
+// GetIdentity looks up identity by ID.
+func (getter *LocalIdentityGetter) GetIdentity(securityIdentity uint64) (*models.Identity, error) {
+	ident := getter.allocator.LookupIdentityByID(context.Background(), identity.NumericIdentity(securityIdentity))
+	if ident == nil {
+		return nil, fmt.Errorf("identity %d not found", securityIdentity)
+	}
+	return ident.GetModel(), nil
+}
+
+// LocalEndpointDNSGetter implements DNSGetter and EndpointGetter interfaces from Hubble.
```

Comment:
What is a DNS getter specifically? I don't fully understand how DNSGetter relates to endpoint IPs for example. More description would be useful here.

Reply:
documented getters.go a bit more.